packages/next/src/server/config-shared.ts

```diff
@@ -499,6 +499,13 @@ export interface ExperimentalConfig {
    */
   typedRoutes?: boolean
 
+  /**
+   * Enable new type-safe Link component with path, params, and searchParams props.
+   * This is a replacement for the existing typedRoutes feature with improved ergonomics.
+   * @see https://nextjs.org/docs/app/api-reference/next-config-js/newTypedRoutes
+   */
+  newTypedRoutes?: boolean
```

Comment:
I think we can keep using the same flag as before as we don't wanna maintain the old version along with this version, it's also experimental so a breaking change in behavior is valid.